File: snap.pl
#!/usr/bin/perl
use strict;
use warnings;
use File::Slurp;
my ($FILE) = @ARGV;
open(my $fh, "<:raw", $FILE);
my ($w,$h) = map int, split ",", read_file("/sys/class/graphics/fb0/virtual_size");
print "P3
${w} ${h}
255
";
sub pr { my ($b, $g, $r, $a)=@_; print "$r $g $b\n"; }
while (read($fh,my $l, 4)) { pr unpack "C4", $l }
close($fh);

File: snap.py
#!/usr/bin/env python3
from struct import unpack
w,h = map( int,open("/sys/class/graphics/fb0/virtual_size").read().split(","))

# returns b g r a
decode = lambda pixel : unpack("4B", pixel)

def pr(b,g,r,a):
    print("%d %d %d" % (r,g,b))

print(f"""P3
{w} {h}
255
""")

with open("/dev/fb0", "rb") as fin:
    while pixel := fin.read(4):
        pr(*decode(pixel))
